src/main/java/org/dependencytrack/policy/cel/CelPolicyLibrary.java

```diff
@@ -143,11 +157,84 @@ public List<ProgramOption> getProgramOptions() {
                                 FUNC_MATCHES_RANGE,
                                 CelPolicyLibrary::matchesRangeFunc
                         ),
-                        Overload.function(FUNC_COMPARE_AGE, CelPolicyLibrary::isComponentOldFunc)
+                        Overload.function(FUNC_COMPARE_AGE,
+                                CelPolicyLibrary::isComponentOldFunc),
+                        Overload.function(FUNC_COMPARE_VERSION_DISTANCE,
+                                CelPolicyLibrary::matchesVersionDistanceFunc)
                 )
         );
     }
 
+    private static Val matchesVersionDistanceFunc(Val... vals) {
+        if (vals.length != 3) {
+            return Types.boolOf(false);
+        }
+        if (vals[0].value() == null || vals[1].value() == null || vals[2].value() == null) {
+            return Types.boolOf(false);
+        }
+
+        if (!(vals[0].value() instanceof final Component component)) {
+            return Err.maybeNoSuchOverloadErr(vals[0]);
+        }
+        if (!(vals[1].value() instanceof final String value)) {
+            return Err.maybeNoSuchOverloadErr(vals[1]);
+        }
+        if (!(vals[2].value() instanceof final String comparator)) {
+            return Err.maybeNoSuchOverloadErr(vals[2]);
+        }
+        if (!component.hasPurl()) {
+            return Err.newErr("Provided component does not have a purl", vals[0]);
+        }
+        try {
+            if (RepositoryType.resolve(new PackageURL(component.getPurl())) == RepositoryType.UNSUPPORTED) {
+                return Err.newErr("Unsupported repository type for component: ", vals[0]);
+            }
+        } catch (MalformedPackageURLException ex) {
+            return Err.newErr("Invalid package url ", component.getPurl());
+        }
+        if (!component.hasLatestVersion()) {
+            return Err.newErr("Requested component does not have latest version information", component);
+        }
+        return Types.boolOf(matchesVersionDIstance(component, value, comparator));
+    }
+
+    private static boolean matchesVersionDIstance(Component component, String value, String comparator) {
+        String comparatorComputed = switch (comparator) {
+            case "NUMERIC_GREATER_THAN", ">" -> "NUMERIC_GREATER_THAN";
+            case "NUMERIC_GREATER_THAN_OR_EQUAL", ">=" -> "NUMERIC_GREATER_THAN_OR_EQUAL";
+            case "NUMERIC_EQUAL", "==" -> "NUMERIC_EQUAL";
+            case "NUMERIC_NOT_EQUAL", "!=" -> "NUMERIC_NOT_EQUAL";
+            case "NUMERIC_LESSER_THAN_OR_EQUAL", "<=" -> "NUMERIC_LESSER_THAN_OR_EQUAL";
+            case "NUMERIC_LESS_THAN", "<" -> "NUMERIC_LESS_THAN";
+            default -> "";
+        };
+        if (comparatorComputed.equals("")) {
+            LOGGER.warn("Was passed a not supported operator for version distance policy " + comparator);
```

Comment:
It would be good to prefix log messages with the function name, that way it's easier to correlate which condition caused it. See the `dependsOn` implementation for reference.